lib/proto.js

```diff
@@ -148,11 +149,15 @@ function write (_obj, msg, num) {
     obj = mixin ? mixin() : {}
   } else {
     obj = Object.assign(mixin ? mixin() : {}, _obj)
-    if (!msg && objError) {
-      msg = _obj.message
-    }
 
     if (objError) {
+      if (!msg) {
+        // replace the main msg with the message of the error
+        msg = _obj.message
+      } else {
+        // if we do have a msg, we should keep the message found on the error
+        obj.message = _obj.message
+      }
```

Comment:
Why? The current implementation doesn't do this, does it?

Reply:
It's true it does not.

Because it does not it results in loosing the `message` string when a `msg` is present, which, in my oppinion is not OK.

But it's true, maybe this should be a separate issue. Should I make a separate one?

Reply:
Yes, please keep issues scoped to one item. It is very difficult to review PRs that address multiple things. However, see https://github.com/pinojs/pino/pull/882

Reply:
I created #889. #882 and  #876 do not seem related to this issue.